### ISSUE TITLE
[Feature Request] Instance target mode only select the instances where the service pod is running

### ISSUE THREAD

**Issue by sczelo** [kubernetes-sigs/aws-load-balancer-controller#3796]
**Is your feature request related to a problem?**

I have a Kubernetes cluster with 60 instances and for every service endpoint (the service uses nodeport and every service uses a different domain) the load balancer controller creates an HTTP Host Header rule and a new target group with all of the 60 instances.
After 17 service, it creates 17 target group with 60 target instance (1020) which is over the 1000 Targets per Application Load Balancer default limit.

**Describe the solution you'd like**

Only insert the instances in the target group, where the pod of the service running. (And update it if the pod moved to another instance)

**Comment by shraddhabang** [comment]
Have you tried using the IP target groups instead Instance target groups?

**Comment by sczelo** [comment]
Yes, but in IP mode it inserts to the target-group the pod kubernetes IP (100.X.X.X) address, not the instance IP address (172.16.X.X).
